packages/snaps-controllers/src/snaps/registry/json.ts

```diff
@@ -346,15 +346,14 @@ export class JsonSnapsRegistry extends BaseController<
   }
 
   /**
-   * Get metadata for the given snap ID.
+   * Get metadata for the given snap ID, if available, without updating registry.
    *
    * @param snapId - The ID of the snap to get metadata for.
    * @returns The metadata for the given snap ID, or `null` if the snap is not
    * verified.
    */
-  async #getMetadata(snapId: string): Promise<SnapsRegistryMetadata | null> {
-    const database = await this.#getDatabase();
-    return database?.verifiedSnaps[snapId]?.metadata ?? null;
+  #getMetadata(snapId: string): SnapsRegistryMetadata | null {
```

Comment:
Technically this is a breaking change, but I don't think we use this anywhere?

cc @Mrtenz

Reply:
Doesn't seem to be used anywhere in the extension outside of the `SnapController` at least, but like you said it's still a breaking change technically. If we want to avoid breaking it, we could keep the function async for now, and pair removing it with another breaking change in the future.

Reply:
IMO not worth it to keep it async, we don't want the performance hit of potentially updating the registry just for analytics

Reply:
I mean keeping the function signature the same, but returning the result without fetching it asynchronously.

Reply:
https://github.com/MetaMask/snaps/pull/3281#discussion_r2039341026